File: spojPractice/spoj4580.java
import java.io.*;
class spoj4580
{
	public static void main(String args[])throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		int a[],b[],c[],d[],e[],f[];
		int n=Integer.parseInt(br.readLine());
		int m[]=new int[n];
		 a=new int[n];
		 b=new int[n];
		 c=new int[n];
		 d=new int[n];
		 e=new int[n];
		 f=new int[n];
		for(int i=0;i<n;i++)
		{
			m[i]=Integer.parseInt(br.readLine());
			a[i]=m[i];
			b[i]=m[i];
			c[i]=m[i];
			d[i]=m[i];
			e[i]=m[i];
			f[i]=m[i];
		}
		long values1[]=new long[(int)Math.pow(n,3)];
		long values2[]=new long[(int)Math.pow(n,3)];
		int count1=0,count2=0;
		for(int i=0;i<n;i++)
		{
			for(int j=0;j<n;j++)
			{
				for(int k=0;k<n;k++)
				{
					values1[count1]=a[i]*b[j]+c[k];
					values2[count2]=d[i]*(e[j]+f[k]);
					if(i!=)
					count1++;
					count2++;
				}
			}
		}
		System.out.println("count1 "+count1);
		System.out.println("count2 "+count2);
		for(int i=0;i<values1.length;i++)
		{
			System.out.print(values1[i]+" ");
		}
		System.out.println();
		for(int i=0;i<values2.length;i++)
		{
			System.out.print(values2[i]+" ");
		}
		System.out.println();
		spoj4580 obj=new spoj4580();
		values1=obj.removeD(values1);
		values2=obj.removeD(values2);
		for(int i=0;i<values1.length;i++)
		{
			System.out.print(values1[i]+" ");
		}
		System.out.println();
		for(int i=0;i<values2.length;i++)
		{
			System.out.print(values2[i]+" ");
		}
		System.out.println();
	}
	public long[] removeD(long a[])
	{
		int n=a.length;
		for(int i=0;i<a.length;i++)
		{
			for(int j=i+1;j<a.length;)
			{
				if(a[i]==a[j])
				{
					for(int k=j;k<a.length;k++)
					{
						a[k]=a[k+1];
						n--;
					}
				}
				else
					j++;
			}
		}
		long ans[]=new long[n];
		for(int i=0;i<n;i++)
		{
			ans[i]=a[i];
		}
		return ans;
	}
}

File: spojPractice/spoj4580.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main()
{
	int *a,*b,*c,*d,*e,*f;
	int n;
	cin>>n;
	int *m;
	m=new int[n];
	a=new int[n];
	b=new int[n];
	c=new int[n];
	d=new int[n];
	e=new int[n];
	f=new int[n];
	
	for(int i=0;i<n;i++)
	{
		cin>>m[i];
		a[i]=m[i];
		b[i]=m[i];
		c[i]=m[i];
		d[i]=m[i];
		e[i]=m[i];
		f[i]=m[i];
	}
	long *values1,*values2;
	values1=new long[(long)pow(n,3)];
	values2=new long[(long)pow(n,3)];
	int count1=0;
	int count2=0;
	for(int i=0;i<n;i++)
	{
		for(int j=0;j<n;j++)
		{
			for(int k=0;k<n;k++)
			{
				values1[count1]=a[i]*b[j]+c[k];
				values2[count2]=d[i]*(e[j]+f[k]);
				count1++;
				count2++;
			}
		}
	}
	cout<<count1<<" cout\n";
	for(int i=0;i<count1;i++)
	{
		cout<<values1[i]<<" ";
	}
	cout<<endl;
	for(int i=0;i<count2;i++)
	{
		cout<<values2[i]<<" ";
	}
	cout<<endl;
	//values1=removeD(values1,count1);
	for(int i=0;i<count1;i++)
	{
		for(int j=i+1;j<count1;)
		{
			if(values1[i]==values1[j])
			{
				for(int k=j;k<count1;k++)
					values1[k]=values1[k+1];
				count1--;
			}
			else
				j++;
		}
	}
	
	//values2=removeD(values2,count2);
	for(int i=0;i<count2;i++)
	{
		for(int j=i+1;j<count2;)
		{
			if(values2[i]==values2[j])
			{
				for(int k=j;k<count2;k++)
					values2[k]=values2[k+1];
				n--;
			}
			else
				j++;
		}
	}
	
	for(int i=0;i<n;i++)
	{
		cout<<values1[i]<<" ";
	}
	cout<<endl;
	for(int i=0;i<n;i++)
	{
		cout<<values2[i]<<" ";
	}
	cout<<endl;
	return 0;
}
/*
		a*b+c=d*(e+f)
*/